source/shared/cpp/ObjectModel/Util.cpp

```diff
@@ -76,43 +76,116 @@ std::string ValidateColor(const std::string& backgroundColor,
+        return;
     }
-    else
+
+    // verify unit
+    const size_t pos = requestedDimension.length() - unit.length();
+    int bresult = requestedDimension.compare(pos, unit.length(), unit);
+    if (bresult != 0)
     {
-        const std::size_t foundIndex = requestedDimension.find(unit);
-        if (std::string::npos == foundIndex || requestedDimension.size() != foundIndex + unit.size())
+        warnings.emplace_back(std::make_shared<AdaptiveCardParseWarning>(
+                AdaptiveSharedNamespace::WarningStatusCode::InvalidDimensionSpecified,
+                warningMessage + requestedDimension)); 
+        return;
+    }
+
+    try
+    {
+        size_t idx;
+        // stoi will get integral value upto non digit char indexed by idx
+        parsedVal = std::stoi(requestedDimension, &idx);
+        if (parsedVal < 0)
         {
-            throw AdaptiveCardParseException(ErrorStatusCode::InvalidPropertyValue,
-                "unit is either missing or inproper form: " + requestedDimension);
+            warnings.emplace_back(std::make_shared<AdaptiveCardParseWarning>(
+                    AdaptiveSharedNamespace::WarningStatusCode::InvalidDimensionSpecified,
+                    warningMessage + requestedDimension)); 
+            parsedVal = 0;
         }
-        try
+
+        const size_t startIndexOfUnit = requestedDimension.length() - unit.size();
+        // if it's decimal point, there should be at least one digit and digits only between decimal point and unit
+        if (unit.length() && requestedDimension.at(idx) == '.' && isdigit(requestedDimension.at(idx + 1)))
         {
-            int parsedVal = std::stoi(requestedDimension.substr(0, foundIndex));
-            if (parsedVal < 0)
-            {
-                throw AdaptiveCardParseException(ErrorStatusCode::InvalidPropertyValue,
-                    "unsigned integer is accepted but received : " + requestedDimension);
-            }
-            parsedDimension = parsedVal;
+            // advance idx up to first non digit char
+            while (idx < startIndexOfUnit && isdigit(requestedDimension.at(++idx)));
         }
-        catch (const std::invalid_argument &e)
+        // non digit char pointed by idx should be the start index of unit
+        if (idx != startIndexOfUnit)
         {
-            (void)e;
-            throw AdaptiveCardParseException(ErrorStatusCode::InvalidPropertyValue,
-                "unsigned integer is accepted but received : " + requestedDimension);
+            warnings.emplace_back(std::make_shared<AdaptiveCardParseWarning>(
+                    AdaptiveSharedNamespace::WarningStatusCode::InvalidDimensionSpecified,
+                    warningMessage + requestedDimension)); 
+            parsedVal = 0;
         }
-        catch (const std::out_of_range &e)
+        parsedDimension = parsedVal;
+    }
+    catch (const std::invalid_argument &e)
+    {
+        (void)e;
```

Comment:
if you don't need `e`, it should probably be left out of the `catch` conditional (i.e. `catch (const std::invalid_argument &)`)

(same below)

Reply:
thanks!

---
In reply to: [217200327](https://github.com/Microsoft/AdaptiveCards/pull/1872#discussion_r217200327) [](ancestors = 217200327)